docs/dev-guide/tscript/messages.mdx

```diff
@@ -0,0 +1,440 @@
+
+Use the [`defineSignal`](https://typescript.temporal.io/api/namespaces/workflow/#definesignal) function to create your Signals:
+
+```ts
+import { defineSignal } from '@temporalio/workflow';
+
+interface JoinInput {
+  userId: string;
+  groupId: string;
+}
+
+export const joinSignal = defineSignal<[JoinInput]>('join');
+```
+
+### Handle a Signal {#handle-signal}
+
+Workflows listen for Signals by the Signal's name.
+Signal handlers are functions defined in the Workflow that listen for incoming Signals of a given type.
+These handlers define how a Workflow should react when it receives a specific type of Signal.
+To implement a Signal handler, you will need to use the [`setHandler`](https://typescript.temporal.io/api/namespaces/workflow/#sethandler) function provided by the TypeScript SDK API.
+
+```ts
+import { setHandler } from '@temporalio/workflow';
+
+export async function yourWorkflow() {
+  const groups = new Map<string, Set<string>>();
+
+  setHandler(joinSignal, ({ userId, groupId }: JoinInput) => {
+    const group = groups.get(groupId);
+    if (group) {
+      group.add(userId);
+    } else {
+      groups.set(groupId, new Set([userId]));
+    }
+  });
+}
+```
+
+The Workflow dynamically reacts to external events without interrupting its execution flow.
+There can be only one handler for any Signal type; calling `setHandler` multiple times for the same Signal type overwrites the existing handler.
+For example, if you first set Handler A for a given Signal, and then later in the Workflow Definition set Handler B for the same Signal, then only Handler B will be invoked upon receiving this Signal.
+
+### Send a Signal from a Temporal Client {#send-signal-from-client}
+
+**How to send a Signal from a Temporal Client**
+
+When a Signal is sent successfully from the Temporal Client, the [WorkflowExecutionSignaled](/references/events#workflowexecutionsignaled) Event appears in the Event History of the Workflow that receives the Signal.
+
+<!--come back and review this and reword-->
+The [`WorkflowHandle.signal`](https://typescript.temporal.io/api/interfaces/client.WorkflowHandle#signal) method synchronously awaits the signal sending?
+
+```typescript
+import { Client } from '@temporalio/client';
+import { joinSignal } from './workflows';
+
+const client = new Client();
+
+const handle = client.workflow.getHandle('workflow-id-123');
+
+await handle.signal(joinSignal, { userId: 'user-1', groupId: 'group-1' });
+```
+
+The getHandle method is used to obtain a reference, or a "handle", to a specific Workflow Execution, which will be used to interact with that Workflow.
+
+### Send a Signal from a Workflow {#send-signal-from-workflow}
+
+A Workflow can send a Signal to another Workflow, in which case it's called an _External Signal_.
+
+When an External Signal is sent:
+
+- A [SignalExternalWorkflowExecutionInitiated](/references/events#signalexternalworkflowexecutioninitiated) Event appears in the sender's Event History.
```

Comment:
We can add 'The`SignalExternalWorkflowExecutionInitiated` Event contains the Signal name as well as a Signal payload. The corresponding Command to this Event is `SignalExternalWorkflowExecution`.'

Reply:
Added